packages/core/src/store/pin-store-factory.ts

```diff
@@ -39,6 +43,12 @@ export class PinStoreFactory {
     const ipfs = this.ipfs
     const pinning = PinningAggregation.build(ipfs, this.pinningEndpoints, this.pinningBackends)
     const retrieve = async (cid: CID): Promise<any> => {
+      const fromCache = this.ipldRecordsCache.get(cid)
```

Comment:
Ideally at some point I'd like to pull out the parts of the Dispatcher that just manage direct puts and gets into IPFS (and add the caching, timeout, and retry logic) into a separate IpfsService API and then use that here directly so we can pick have unified access to the caching logic and also pick up the other extras that the Dispatcher adds over raw IPFS access.  But that is certainly out of scope for this PR.

Reply:
Someday, yeah. That’s on my list too.